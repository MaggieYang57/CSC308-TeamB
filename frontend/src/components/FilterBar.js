import React, { Component, useState, useEffect } from "react";
import axios from "axios";
import "../css/filterBar.css";
import FilterButton from "./FilterButton";
import SortDrop from "./SortDrop";
import SearchBar from "./SearchBar";

function FilterBar(props) {

  return (
    <div id="filterBar">
      <SearchBar />
      <SortDrop onChange={props.onDiff}/>
      
      <div id="filter-buttons">
        <FilterButton name="Dog Friendly" filterType='dog_friendly' onChange={props.onChange} />
        <FilterButton name="Family Friendly" filterType='family_friendly' onChange={props.onChange}/>
        <FilterButton name="Horseback Riding" filterType='horseback_riding' onChange={props.onChange}/>
        <FilterButton name="Bike Riding" filterType='biking' onChange={props.onChange}/>
      </div>
    </div>
  );
  
}

export default FilterBar;
