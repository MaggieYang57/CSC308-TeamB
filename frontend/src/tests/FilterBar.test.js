import React from 'react';
import renderer from 'react-test-renderer';
import FilterBar from '../components/FilterBar';

describe("FilterBar", () => {

  it('renders correctly', () => {
    const tree = renderer
      .create(<FilterBar />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

})