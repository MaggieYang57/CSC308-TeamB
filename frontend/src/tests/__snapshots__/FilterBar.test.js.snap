// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBar renders correctly 1`] = `
<div
  id="filterBar"
>
  <input
    onChange={[Function]}
    placeholder="Search Hikes"
    style={
      Object {
        "background": "#F2F1F9",
        "border": "none",
        "marginTop": "4vw",
        "padding": "0.5rem",
        "width": "20rem",
      }
    }
    value=""
  />
  <div
    id="sort-drop"
  >
    <form>
      <label
        htmlFor="difficulty"
      >
        Sort Hikes by:
      </label>
      <select
        id="difficulty"
        name="difficulty"
        onChange={[Function]}
      >
        <option
          value="0"
        >
          None
        </option>
        <option
          value="2"
        >
          Difficulty: Low to High
        </option>
        <option
          value="1"
        >
          Difficulty: High to Low
        </option>
      </select>
    </form>
  </div>
  <div
    id="filter-buttons"
  >
    <div
      className="filterButton"
    >
      <label>
        <input
          onClick={[Function]}
          type="checkbox"
          value="1"
        />
        <span>
          Dog Friendly
        </span>
      </label>
    </div>
    <div
      className="filterButton"
    >
      <label>
        <input
          onClick={[Function]}
          type="checkbox"
          value="1"
        />
        <span>
          Family Friendly
        </span>
      </label>
    </div>
    <div
      className="filterButton"
    >
      <label>
        <input
          onClick={[Function]}
          type="checkbox"
          value="1"
        />
        <span>
          Horseback Riding
        </span>
      </label>
    </div>
    <div
      className="filterButton"
    >
      <label>
        <input
          onClick={[Function]}
          type="checkbox"
          value="1"
        />
        <span>
          Bike Riding
        </span>
      </label>
    </div>
  </div>
</div>
`;
